scripts/monitoring/auto_monitor_scale_workers.py

```diff
@@ -0,0 +1,404 @@
+            {"Name": "ClusterName", "Value": cluster_name},
+            {"Name": "ServiceName", "Value": service_name},
+        ],
+        StartTime=start_time,
+        EndTime=end_time,
+        Period=period_seconds,
+        Statistics=["Average"],
+    )
+
+    return response["Datapoints"]
+
+
+# get current CPU and Memory utilization for a challenge
+def current_worker_limit(task_definition_name, metrics):
+    ecs_client = get_boto3_client("ecs", aws_keys)
+    try:
+        response = ecs_client.describe_task_definition(
+            taskDefinition=task_definition_name
+        )
+    except Exception as e:
+        print(f"Error retrieving task definition: {str(e)}")
+        return {}
+
+    task_definition = response.get("taskDefinition", {})
+    return task_definition.get(metrics, 0)
+
+
+def get_new_resource_limit(
+    metrics, metric_name, service_name, current_metric_limit
+):
+    """Categorize CPU or Memory metrics and decide on scaling actions separately."""
+
+    # Calculate average of the provided metrics
+    average = sum(
+        datapoint["Average"] for datapoint in metrics if "Average" in datapoint
+    ) / len(metrics)
+
+    # Apply separate logic based on whether the metric is CPU or Memory
+    if metric_name == "CPU":
+        # CPU-specific scaling logic
+        if average <= 25:
+            # if average smaller than 25%, scale down
+            print(
+                f"Scaling down {service_name} due to low {metric_name} utilization"
+            )
+            new_limit = str(int(current_metric_limit) // 2)
+        elif average >= 75:
+            # if average greater than 75%, scale up
+            print(
+                f"Scaling up {service_name} due to high {metric_name} utilization"
+            )
+            new_limit = str(int(current_metric_limit) * 2)
+        else:
+            # no scaling action required
+            print(
+                f"No scaling action required for {service_name} based on {metric_name} utilization"
+            )
+            new_limit = current_metric_limit
+        return new_limit
+    elif metric_name == "Memory":
+        if average <= 25:
+            # if average smaller than 25%, scale down
+            print(
+                f"Scaling down {service_name} due to low {metric_name} utilization"
+            )
+            new_limit = str(int(current_metric_limit) // 2)
+        elif average >= 75:
+            # if average greater than 75%, scale up
+            print(
+                f"Scaling up {service_name} due to high {metric_name} utilization"
+            )
+            new_limit = str(int(current_metric_limit) * 2)
+        else:
+            # no scaling action required
+            print(
+                f"No scaling action required for {service_name} based on {metric_name} utilization"
+            )
+            new_limit = current_metric_limit
+        return new_limit
+
+
+def get_task_definition_details(task_definition_name):
```

Comment:
This function extracts the current settings from the task definition (which includes CPU, Memory, Storage, logging configurations, etc.). It retrieves the current values, updates the limit, and then saves the modified settings.